### ISSUE TITLE
New Centos9 base image is missing dependencies needed for pulp_rpm's createrepo_c dependency

### ISSUE THREAD

**Issue by gerrod3** [pulp/pulp-oci-images#601]
**Version**
Latest centos9 base image

**Describe the bug**
When using the latest image with pulp_rpm installed, anytime a pulp instance is started (anytime the pulp_rpm plugin is loaded) an error message, "Failed to initialize NSS library", is printed to stderr. 

**To Reproduce**
Use the latest image with pulp_rpm installed. In the container's shell try `createrepo_c -h` and see the error message at the top of the output.

**Comment by dralley** [comment]
This is interesting.

So, NSS isn't even present on the base image, for either CentOS 9 Stream OR CentOS 8 Stream OR either version of our base image.

It is, however, injected into the `createrepo_c` PyPI package during the build process, which you can see when you install and uninstall it:

```
[root@734e05321fa8 /]# python3.9 -m pip uninstall createrepo_c
Found existing installation: createrepo-c 1.0.4
Uninstalling createrepo-c-1.0.4:
  Would remove:
    /usr/local/bin/createrepo_c
    /usr/local/bin/mergerepo_c
    /usr/local/bin/modifyrepo_c
    /usr/local/bin/sqliterepo_c
    /usr/local/lib64/python3.9/site-packages/createrepo_c-1.0.4.dist-info/*
    /usr/local/lib64/python3.9/site-packages/createrepo_c.libs/libacl-e4e61e13.so.1.1.0
    /usr/local/lib64/python3.9/site-packages/createrepo_c.libs/libattr-4f2a9577.so.1.1.0
    /usr/local/lib64/python3.9/site-packages/createrepo_c.libs/libaudit-36f6f84a.so.1.0.0
    /usr/local/lib64/python3.9/site-packages/createrepo_c.libs/libblkid-fccc232f.so.1.1.0
    /usr/local/lib64/python3.9/site-packages/createrepo_c.libs/libbz2-a273e504.so.1.0.6
    /usr/local/lib64/python3.9/site-packages/createrepo_c.libs/libcap-79733b59.so.2.22
    /usr/local/lib64/python3.9/site-packages/createrepo_c.libs/libcap-ng-2c83fa0e.so.0.0.0
    /usr/local/lib64/python3.9/site-packages/createrepo_c.libs/libcom_err-2abe824b.so.2.1
    /usr/local/lib64/python3.9/site-packages/createrepo_c.libs/libcrypt-8f974802.so.1
    /usr/local/lib64/python3.9/site-packages/createrepo_c.libs/libcrypto-d3570994.so.1.0.2k
    /usr/local/lib64/python3.9/site-packages/createrepo_c.libs/libcurl-34c6b10f.so.4.3.0
    /usr/local/lib64/python3.9/site-packages/createrepo_c.libs/libdb-5-a712f772.3.so
    /usr/local/lib64/python3.9/site-packages/createrepo_c.libs/libelf-0-3d9de92a.176.so
    /usr/local/lib64/python3.9/site-packages/createrepo_c.libs/libffi-af4ed708.so.6.0.1
    /usr/local/lib64/python3.9/site-packages/createrepo_c.libs/libgio-2-8dd0156d.0.so.0.5600.1
    /usr/local/lib64/python3.9/site-packages/createrepo_c.libs/libgmodule-2-ef3ece87.0.so.0.5600.1
    /usr/local/lib64/python3.9/site-packages/createrepo_c.libs/libgssapi_krb5-497db0c6.so.2.2
    /usr/local/lib64/python3.9/site-packages/createrepo_c.libs/libidn-97d26f25.so.11.6.11
    /usr/local/lib64/python3.9/site-packages/createrepo_c.libs/libk5crypto-b1f99d5c.so.3.1
    /usr/local/lib64/python3.9/site-packages/createrepo_c.libs/libkeyutils-dfe70bd6.so.1.5
    /usr/local/lib64/python3.9/site-packages/createrepo_c.libs/libkrb5-fcafa220.so.3.3
    /usr/local/lib64/python3.9/site-packages/createrepo_c.libs/libkrb5support-d0bcff84.so.0.1
    /usr/local/lib64/python3.9/site-packages/createrepo_c.libs/liblber-2-7eefa761.4.so.2.10.7
    /usr/local/lib64/python3.9/site-packages/createrepo_c.libs/libldap-2-b8523ca2.4.so.2.10.7
    /usr/local/lib64/python3.9/site-packages/createrepo_c.libs/liblua-5-37c36ff7.1.so
    /usr/local/lib64/python3.9/site-packages/createrepo_c.libs/liblzma-004595ca.so.5.2.2
    /usr/local/lib64/python3.9/site-packages/createrepo_c.libs/libmodulemd-67c941ba.so.2.15.0
    /usr/local/lib64/python3.9/site-packages/createrepo_c.libs/libmount-2c8ae47c.so.1.1.0
    /usr/local/lib64/python3.9/site-packages/createrepo_c.libs/libnspr4-4f93563e.so
    /usr/local/lib64/python3.9/site-packages/createrepo_c.libs/libnss3-4d0ee2f8.so
    /usr/local/lib64/python3.9/site-packages/createrepo_c.libs/libnssutil3-5292dbaa.so
    /usr/local/lib64/python3.9/site-packages/createrepo_c.libs/libpcre-9513aab5.so.1.2.0
    /usr/local/lib64/python3.9/site-packages/createrepo_c.libs/libplc4-4e123408.so
    /usr/local/lib64/python3.9/site-packages/createrepo_c.libs/libplds4-f0f0f720.so
    /usr/local/lib64/python3.9/site-packages/createrepo_c.libs/libpopt-5fc6f60f.so.0.0.0
    /usr/local/lib64/python3.9/site-packages/createrepo_c.libs/librpm-61c7936b.so.3.2.2
    /usr/local/lib64/python3.9/site-packages/createrepo_c.libs/librpmio-4a63bcce.so.3.2.2
    /usr/local/lib64/python3.9/site-packages/createrepo_c.libs/libsasl2-84219a89.so.3.0.0
    /usr/local/lib64/python3.9/site-packages/createrepo_c.libs/libselinux-0922c95c.so.1
    /usr/local/lib64/python3.9/site-packages/createrepo_c.libs/libsmime3-985a6d0a.so
    /usr/local/lib64/python3.9/site-packages/createrepo_c.libs/libsqlite3-a3ea561f.so.0.8.6
    /usr/local/lib64/python3.9/site-packages/createrepo_c.libs/libssh2-1408df5a.so.1.0.1
    /usr/local/lib64/python3.9/site-packages/createrepo_c.libs/libssl-cd1d6220.so.1.0.2k
    /usr/local/lib64/python3.9/site-packages/createrepo_c.libs/libssl3-7d7debcb.so
    /usr/local/lib64/python3.9/site-packages/createrepo_c.libs/libuuid-f64cda11.so.1.3.0
    /usr/local/lib64/python3.9/site-packages/createrepo_c.libs/libxml2-3998bec4.so.2.9.
    /usr/local/lib64/python3.9/site-packages/createrepo_c.libs/libyaml-0-1bf300b2.so.2.0.4
    /usr/local/lib64/python3.9/site-packages/createrepo_c.libs/libzstd-8d126cbb.so.1.5.5
    /usr/local/lib64/python3.9/site-packages/createrepo_c/*
Proceed (y/n)? 
```

And I can reproduce this issue on both by simply installing createrepo_c from source and running `createrepo_c --version`.  It can also be reproduced by opening a Python shell and simply importing the library.

However, even though it prints to stderr, it doesn't seem to actually exit or anything?  Importing `createrepo_c` while it does print the error doesn't seem to raise any exceptions or crash the process?

Anyway, running strace when `createrepo_c --version` executes does show that it does gets into some DNS resolution setup and then fails when attempting to open `/usr/lib64/libsoftokn3.so`.  I guess it's not a fatal error, though, because it keeps going afterwards.

```
openat(AT_FDCWD, "/usr/lib64/avx512_1/libnss_sss.so.2", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
newfstatat(AT_FDCWD, "/usr/lib64/avx512_1", 0x7ffee135e160, 0) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/usr/lib64/x86_64/libnss_sss.so.2", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
newfstatat(AT_FDCWD, "/usr/lib64/x86_64", 0x7ffee135e160, 0) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/usr/lib64/libnss_sss.so.2", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
newfstatat(AT_FDCWD, "/usr/lib64", {st_mode=S_IFDIR|0555, st_size=482, ...}, 0) = 0
munmap(0x7fa0dbc4c000, 7219)            = 0
openat(AT_FDCWD, "/etc/passwd", O_RDONLY|O_CLOEXEC) = 3
fstat(3, {st_mode=S_IFREG|0644, st_size=589, ...}) = 0
lseek(3, 0, SEEK_SET)                   = 0
read(3, "root:x:0:0:root:/root:/bin/bash\n"..., 4096) = 589
close(3)                                = 0
socket(AF_UNIX, SOCK_STREAM|SOCK_CLOEXEC|SOCK_NONBLOCK, 0) = 3
connect(3, {sa_family=AF_UNIX, sun_path="/var/run/nscd/socket"}, 110) = -1 ENOENT (No such file or directory)
close(3)                                = 0
socket(AF_UNIX, SOCK_STREAM|SOCK_CLOEXEC|SOCK_NONBLOCK, 0) = 3
connect(3, {sa_family=AF_UNIX, sun_path="/var/run/nscd/socket"}, 110) = -1 ENOENT (No such file or directory)
close(3)                                = 0
newfstatat(AT_FDCWD, "/etc/nsswitch.conf", {st_mode=S_IFREG|0644, st_size=2108, ...}, 0) = 0
openat(AT_FDCWD, "/etc/group", O_RDONLY|O_CLOEXEC) = 3
fstat(3, {st_mode=S_IFREG|0644, st_size=274, ...}) = 0
lseek(3, 0, SEEK_SET)                   = 0
read(3, "root:x:0:\nbin:x:1:\ndaemon:x:2:\ns"..., 4096) = 274
close(3)                                = 0
newfstatat(AT_FDCWD, "/etc/resolv.conf", {st_mode=S_IFREG|0644, st_size=43, ...}, 0) = 0
openat(AT_FDCWD, "/etc/host.conf", O_RDONLY|O_CLOEXEC) = 3
fstat(3, {st_mode=S_IFREG|0644, st_size=9, ...}) = 0
read(3, "multi on\n", 4096)             = 9
read(3, "", 4096)                       = 0
close(3)                                = 0
futex(0x7fa0db8e348c, FUTEX_WAKE_PRIVATE, 2147483647) = 0
openat(AT_FDCWD, "/etc/resolv.conf", O_RDONLY|O_CLOEXEC) = 3
fstat(3, {st_mode=S_IFREG|0644, st_size=43, ...}) = 0
read(3, "search lan rht.csb\nnameserver 19"..., 4096) = 43
read(3, "", 4096)                       = 0
fstat(3, {st_mode=S_IFREG|0644, st_size=43, ...}) = 0
close(3)                                = 0
socket(AF_UNIX, SOCK_STREAM|SOCK_CLOEXEC|SOCK_NONBLOCK, 0) = 3
connect(3, {sa_family=AF_UNIX, sun_path="/var/run/nscd/socket"}, 110) = -1 ENOENT (No such file or directory)
close(3)                                = 0
socket(AF_UNIX, SOCK_STREAM|SOCK_CLOEXEC|SOCK_NONBLOCK, 0) = 3
connect(3, {sa_family=AF_UNIX, sun_path="/var/run/nscd/socket"}, 110) = -1 ENOENT (No such file or directory)
close(3)                                = 0
newfstatat(AT_FDCWD, "/etc/nsswitch.conf", {st_mode=S_IFREG|0644, st_size=2108, ...}, 0) = 0
openat(AT_FDCWD, "/etc/hosts", O_RDONLY|O_CLOEXEC) = 3
fstat(3, {st_mode=S_IFREG|0644, st_size=244, ...}) = 0
lseek(3, 0, SEEK_SET)                   = 0
read(3, "127.0.0.1\tlocalhost localhost.lo"..., 4096) = 244
read(3, "", 4096)                       = 0
close(3)                                = 0
gettid()                                = 104
rt_sigaction(SIGPIPE, {sa_handler=SIG_IGN, sa_mask=[], sa_flags=SA_RESTORER, sa_restorer=0x7fa0db71e6f0}, NULL, 8) = 0
openat(AT_FDCWD, "/proc/sys/crypto/fips_enabled", O_RDONLY) = 3
fstat(3, {st_mode=S_IFREG|0444, st_size=0, ...}) = 0
read(3, "0\n", 1024)                    = 2
close(3)                                = 0
openat(AT_FDCWD, "/usr/local/lib64/python3.9/site-packages/createrepo_c/../createrepo_c.libs/libsoftokn3.so", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
readlink("/usr/local/lib64/python3.9/site-packages/createrepo_c/../createrepo_c.libs/libnss3-4d0ee2f8.so", 0x55e2f2a10c00, 1023) = -1 EINVAL (Invalid argument)
openat(AT_FDCWD, "/usr/local/lib64/python3.9/site-packages/createrepo_c/../createrepo_c.libs/libsoftokn3.so", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/etc/ld.so.cache", O_RDONLY|O_CLOEXEC) = 3
fstat(3, {st_mode=S_IFREG|0644, st_size=7219, ...}) = 0
mmap(NULL, 7219, PROT_READ, MAP_PRIVATE, 3, 0) = 0x7fa0dbc4c000
close(3)                                = 0
openat(AT_FDCWD, "/lib64/libsoftokn3.so", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/usr/lib64/libsoftokn3.so", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
munmap(0x7fa0dbc4c000, 7219)            = 0
write(2, "error: ", 7error: )                  = 7
write(2, "Failed to initialize NSS library"..., 33Failed to initialize NSS library
```

(more stuff, and it does print the version)

Installing the `nss` package however does resolve the error message on both.  I don't understand yet why the CentOS 8 Stream-based OCI environment seems to not have this issue when I can reproduce it directly on CentOS 8 Stream.  And I confirmed that `nss` isn't even installed in our OCI environment.

**Comment by dralley** [comment]
> I don't understand yet why the CentOS 8 Stream-based OCI environment seems to not have this issue when I can reproduce it directly on CentOS 8 Stream.

It's because I was using Python 3.9 on CentOS 8 Stream when I tested it.  Had I used Python 3.8 like the Pulp base image it would have had the same issue.

The difference is that Python 3.9+ have prebuilt wheels available on PyPI and Python 3.8 and previous do not.  So on Python 3.8 it was being built from source, and on Python 3.9 it was using the prebuilt wheel.

Strictly speaking this isn't an oci image / environment issue.
